### PR TITLE
add .gpg url (match github behaviour)

### PR DESCRIPTION
**Pull request by sapk**
After some testing of #4193 I added what was missing to have it working. Thx to @techknowlogick that lay down most of the bases.

Basically, we haven't in database what is needed to fully reconstruct the gpg armored export (ex: signature, full identities, ...). 
I add a table containing the imported armored key that we can re-used to be join with others imported keys and exported.
Like github previous keys that need re-import are marked as a header (ex: https://github.com/sapk.gpg)

The only bug is that the cipherTypeByte are changed by the export by golang compared to the imported one but that doesn't change the interpretation of the gpg armored key.

### CODE REVIEW

**Comment by techknowlogick** [comment]
@sapk Thanks for continuing this PR 😃

File: models/models.go [hunk 1]
```diff
@@ -108,6 +108,7 @@ func init() {
 		new(LFSMetaObject),
 		new(TwoFactor),
 		new(GPGKey),
+		new(GPGKeyImport),
```
Comment (techknowlogick) [inline models/models.go:111]: Can you add a migration to add this struct to DB (I know the sync below auto does this, however sync runs after migrations and could run into problems)

File: routers/user/profile.go [hunk 2]
```diff
@@ -59,9 +59,16 @@ func Profile(ctx *context.Context) {
 	isShowKeys := false
 	if strings.HasSuffix(uname, ".keys") {
 		isShowKeys = true
+		uname = strings.TrimSuffix(uname, ".gpg")
```
Comment (techknowlogick) [inline routers/user/profile.go]:
```suggestion
		uname = strings.TrimSuffix(uname, ".keys")
```

**Comment by sapk** [comment]
For info, [imported](https://cirw.in/gpg-decoder/#-----BEGIN%20PGP%20PUBLIC%20KEY%20BLOCK-----%0A%0AmQENBFqN9A0BCAC2ViMyOmf1uzNZOHDh7gJpw2/99DazxAdt1YAo94t61rPQybb9%0AyxwXuybsVKRBHQOm6eDE8icVDbzWGuspBiGpT1VcANPUDc2NatPidypjeVA+3ycp%0Aac9kqeW/iAfyPszopQ2gtRTafkJAx71zB6NkemPCB8JGAgNyvbRRvHYm3rGBU93m%0Ac0OYE1wfHnkmDoTBgNImhMnalg5NU1V+KQfWiYCH9iZ8OPBLHSkIeEpdCHW/6ey8%0AcqKCTkbnoDzwdC31djD8Tftzj8ys8vZtZe4No1FuP0Gqt3j+JwBKMbXyVMZ9BiQA%0ALxATJS5c3PMnjXnR/AZFuv7VDX7gRWzTGQ4VABEBAAG0HUFudG9pbmUgR0lSQVJE%0AIDxzYXBrQHNhcGsuZnI+iQFUBBMBCAA+FiEE+W2OHhUHqwGWFFr987RtFwjiKWMF%0AAlqN9A0CGwMFCQPCZwAFCwkIBwIGFQoJCAsCBBYCAwECHgECF4AACgkQ87RtFwji%0AKWOP6AgAtfbdkhSul2E2IuviavMNF8cJwuDtOdBFXA8maJFfTu6p55Qz7IPPmFzW%0ADYv2X4VM/bYRCzu0S2kVQ49kWvBT12XhV/wLdM9nCPSJhxyeQtS82w8o8NTOJKuq%0AJx2jcGsymtxotjoQWooG0oye10K3WFtb2yUr7HNePlv52q46RhNYkMGsIKDnpIlR%0AcYHeZLbuJ3bCllfrglMBSY7pd7HRlLF/+ho7LHIW8JhZqvneq9hlcBjLfDC7IZY2%0AxH/5qCbrjd9tdBVJAGrv1nhsRkqaGPBvWWb+G3J5JPWhjdVk12RqxEw5fuW2wknd%0At72Ep2qujsE8S1h2VdA51emTYt4PY7kBDQRajfQNAQgA3xBXPNH0Rs2Ad3CHHDl9%0AZToz5E2+YGIDTrLA0BbNLZrszDm2ETVyvWCehrgcJ7ILIh5MVNZ9JzZPOIO8nzOU%0AQoWq20jjJvHUBwUenpvlNSmMOZ+cX1z+Xj7R/nC0TTZ8Ily617fgRpwLvpg8bQkT%0AG+BYypx96nBnglo/2R++xnFb7iIMRANyTy4a0wrryb4PAhwPbNk79QS/vMq+MCr/%0A58IJMdNPZHmcZRM4VmXGuQ8nf2Hfjg0wg/28yMd+A8tXWKTPUQrjF3xHnPI60DQv%0AC93oFO97AHagSlNQ3Z1rnyQjUqeZuk7WbuvOIVWaSjkSd92Sj0HPbe0rrlatq/pD%0AFQARAQABiQE8BBgBCAAmFiEE+W2OHhUHqwGWFFr987RtFwjiKWMFAlqN9A0CGwwF%0ACQPCZwAACgkQ87RtFwjiKWOH5Af/QkrY2YaZ9onyfVFc9NVQuuOGQHrnXm7ZoiYD%0AbPOiQc8qJwH74JzQnPJoM/qdGq3soHFebsiJ/uWZ6aPvQ6QPUnZldA21aEdRE5Cg%0APN5hyUmzitKaJx1f5k//G5xTZPdokqTdArygdTgNME6XsmvTCuGizEj9koH3LU/G%0AbLeC+urKouPohJawKd2jFvXWbbk9UIbZCB1P2Dj/2+LFOWr1FGV/YCJq5TxBfShm%0AZmeUnYQPZWyLi6VwuqwTu8NGP1U0fA+ud6SGxNiPBdfxusZ2QWl+Ps1AQyovRi5C%0AeQm0qC0/wtaYahmp9vfWlBoMKgRewxUsI3GgpKK0Nt4w1wq3IA==%0A=DfyY%0A-----END%20PGP%20PUBLIC%20KEY%20BLOCK-----%0A) vs [exported](https://cirw.in/gpg-decoder/#-----BEGIN%20PGP%20PUBLIC%20KEY%20BLOCK-----%0A%0AxsBNBFqN9A0BCAC2ViMyOmf1uzNZOHDh7gJpw2/99DazxAdt1YAo94t61rPQybb9%0AyxwXuybsVKRBHQOm6eDE8icVDbzWGuspBiGpT1VcANPUDc2NatPidypjeVA+3ycp%0Aac9kqeW/iAfyPszopQ2gtRTafkJAx71zB6NkemPCB8JGAgNyvbRRvHYm3rGBU93m%0Ac0OYE1wfHnkmDoTBgNImhMnalg5NU1V+KQfWiYCH9iZ8OPBLHSkIeEpdCHW/6ey8%0AcqKCTkbnoDzwdC31djD8Tftzj8ys8vZtZe4No1FuP0Gqt3j+JwBKMbXyVMZ9BiQA%0ALxATJS5c3PMnjXnR/AZFuv7VDX7gRWzTGQ4VABEBAAHNHUFudG9pbmUgR0lSQVJE%0AIDxzYXBrQHNhcGsuZnI+wsCUBBMBCAA+FiEE+W2OHhUHqwGWFFr987RtFwjiKWMF%0AAlqN9A0CGwMFCQPCZwAFCwkIBwIGFQoJCAsCBBYCAwECHgECF4AACgkQ87RtFwji%0AKWOP6AgAtfbdkhSul2E2IuviavMNF8cJwuDtOdBFXA8maJFfTu6p55Qz7IPPmFzW%0ADYv2X4VM/bYRCzu0S2kVQ49kWvBT12XhV/wLdM9nCPSJhxyeQtS82w8o8NTOJKuq%0AJx2jcGsymtxotjoQWooG0oye10K3WFtb2yUr7HNePlv52q46RhNYkMGsIKDnpIlR%0AcYHeZLbuJ3bCllfrglMBSY7pd7HRlLF/+ho7LHIW8JhZqvneq9hlcBjLfDC7IZY2%0AxH/5qCbrjd9tdBVJAGrv1nhsRkqaGPBvWWb+G3J5JPWhjdVk12RqxEw5fuW2wknd%0At72Ep2qujsE8S1h2VdA51emTYt4PY87ATQRajfQNAQgA3xBXPNH0Rs2Ad3CHHDl9%0AZToz5E2+YGIDTrLA0BbNLZrszDm2ETVyvWCehrgcJ7ILIh5MVNZ9JzZPOIO8nzOU%0AQoWq20jjJvHUBwUenpvlNSmMOZ+cX1z+Xj7R/nC0TTZ8Ily617fgRpwLvpg8bQkT%0AG+BYypx96nBnglo/2R++xnFb7iIMRANyTy4a0wrryb4PAhwPbNk79QS/vMq+MCr/%0A58IJMdNPZHmcZRM4VmXGuQ8nf2Hfjg0wg/28yMd+A8tXWKTPUQrjF3xHnPI60DQv%0AC93oFO97AHagSlNQ3Z1rnyQjUqeZuk7WbuvOIVWaSjkSd92Sj0HPbe0rrlatq/pD%0AFQARAQABwsB8BBgBCAAmFiEE+W2OHhUHqwGWFFr987RtFwjiKWMFAlqN9A0CGwwF%0ACQPCZwAACgkQ87RtFwjiKWOH5Af/QkrY2YaZ9onyfVFc9NVQuuOGQHrnXm7ZoiYD%0AbPOiQc8qJwH74JzQnPJoM/qdGq3soHFebsiJ/uWZ6aPvQ6QPUnZldA21aEdRE5Cg%0APN5hyUmzitKaJx1f5k//G5xTZPdokqTdArygdTgNME6XsmvTCuGizEj9koH3LU/G%0AbLeC+urKouPohJawKd2jFvXWbbk9UIbZCB1P2Dj/2+LFOWr1FGV/YCJq5TxBfShm%0AZmeUnYQPZWyLi6VwuqwTu8NGP1U0fA+ud6SGxNiPBdfxusZ2QWl+Ps1AQyovRi5C%0AeQm0qC0/wtaYahmp9vfWlBoMKgRewxUsI3GgpKK0Nt4w1wq3IA==%0A=jrN6%0A-----END%20PGP%20PUBLIC%20KEY%20BLOCK-----%0A) key

File: models/models.go (hunk 1 above)
Comment (sapk) [reply models/models.go:111]: @techknowlogick To understand you want me to do. I just need to add a  migrations script only sync the new struct ?

Comment (sapk) [reply models/models.go:111]: Looking at just the previous migration should be done.

Comment (techknowlogick) [reply models/models.go:111]: Yes, just a sync2. The most recent migration is an excellent example.

Comment (techknowlogick) [reply models/models.go:111]: Ah, I just refreshed and saw you pushed the change. Thanks :)

**Comment by lunny** [comment]
@sapk @techknowlogick good work. With integration tests are better.

**Comment by sapk** [comment]
@lunny done

**Review by lunny — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**